WalletWasabi.Fluent/ViewModels/Wallets/LoadingViewModel.cs

```diff
@@ -100,12 +103,31 @@ private async Task LoadWalletAsync(bool isBackendAvailable)
 
 		await SetInitValuesAsync(isBackendAvailable).ConfigureAwait(false);
 
-		while (isBackendAvailable && RemainingFiltersToDownload > 0 && !_wallet.KeyManager.SkipSynchronization)
+		while (isBackendAvailable && RemainingFiltersToDownload > 0 && !_nbwsvm.Wallet.KeyManager.SkipSynchronization)
 		{
 			await Task.Delay(1000).ConfigureAwait(false);
 		}
 
-		await UiServices.WalletManager.LoadWalletAsync(_wallet).ConfigureAwait(false);
+		if (_nbwsvm.Wallet.State != WalletState.Uninitialized)
+		{
+			throw new Exception("Wallet is already being logged in.");
+		}
+
+		try
+		{
+			await Task.Run(async () => await Services.WalletManager.StartWalletAsync(_nbwsvm.Wallet));
+
+			_nbwsvm.WalletViewModel = WalletViewModel.Create(UiContext, _nbwsvm);
+			_nbwsvm.CurrentPage = _nbwsvm.WalletViewModel;
+		}
+		catch (OperationCanceledException ex)
```

Comment:
Useless since there's an universal catch block.

Reply:
Note taken. Will make sure to fix in later PR which refactors this class